arangod/Graph/PathManagement/PathResult.cpp

```diff
@@ -165,6 +169,45 @@ auto PathResult<ProviderType, Step>::isEmpty() const -> bool {
   return _vertices.empty();
 }
 
+template<class ProviderType, class Step>
+auto PathResult<ProviderType, Step>::getMemoryUsage() const -> size_t {
+  size_t mem = 0;
+  mem += sizeof(PathResult<ProviderType, Step>);
+  mem += sizeof(typename Step::Vertex) * _vertices.size();
+  mem += sizeof(typename Step::Edge) * _edges.size();
+  mem += sizeof(double) * _weights.size();
+  return mem;
+}
+
+template<class ProviderType, class Step>
+auto PathResult<ProviderType, Step>::compare(
+    PathResult<ProviderType, Step> const& other) -> std::strong_ordering {
+  if (_pathWeight > other._pathWeight) {
+    // The ">" here is intentional! We want descending weight!
+    return std::strong_ordering::less;
+  }
+  if (_pathWeight < other._pathWeight) {
+    return std::strong_ordering::greater;
+  }
+  // Now let's take the len-lexicographic ordering of the edges. We do not
+  // have to consider the vertices, since if the edges are equal, then
+  // their end vertices are equal:
+  if (_edges.size() < other._edges.size()) {
+    return std::strong_ordering::less;
+  }
+  if (_edges.size() > other._edges.size()) {
+    return std::strong_ordering::greater;
+  }
```

Comment:
I would prefer using spaceship operator, but I understand that is a bit bigger changes, so just here is fine
```suggestion
  if (_edges.size() != other._edges.size()) {
    return _edges.size() <=> other._edges.size();
  }
```